contracts/contracts/strategies/balancer/BaseAuraStrategy.sol

```diff
@@ -0,0 +1,140 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
+
+/**
+ * @title OETH Base Balancer Abstract Strategy
+ * @author Origin Protocol Inc
+ */
+
+import { BaseBalancerStrategy } from "./BaseBalancerStrategy.sol";
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import { IRateProvider } from "../../interfaces/balancer/IRateProvider.sol";
+import { IERC20 } from "../../utils/InitializableAbstractStrategy.sol";
+import { IERC4626 } from "../../../lib/openzeppelin/interfaces/IERC4626.sol";
+import { StableMath } from "../../utils/StableMath.sol";
+import { IRewardStaking } from "../IRewardStaking.sol";
+
+abstract contract BaseAuraStrategy is BaseBalancerStrategy {
+    using SafeERC20 for IERC20;
+    using StableMath for uint256;
+
+    /// @notice Address of the Aura rewards pool
+    address public immutable auraRewardPoolAddress;
+
+    // renamed from __reserved to not shadow BaseBalancerStrategy.__reserved,
+    int256[50] private __reserved_2;
+
+    constructor(address _auraRewardPoolAddress) {
+        auraRewardPoolAddress = _auraRewardPoolAddress;
+    }
+
+    /**
+     * Initializer for setting up strategy internal state. This overrides the
+     * InitializableAbstractStrategy initializer as Balancer's strategies don't fit
+     * well within that abstraction.
+     * @param _rewardTokenAddresses Address of BAL & AURA
+     * @param _assets Addresses of supported assets. MUST be passed in the same
+     *                order as returned by coins on the pool contract, i.e.
+     *                WETH, stETH
+     * @param _pTokens Platform Token corresponding addresses
+     */
+    function initialize(
+        address[] calldata _rewardTokenAddresses, // BAL & AURA
+        address[] calldata _assets,
+        address[] calldata _pTokens
+    ) external override onlyGovernor initializer {
+        maxWithdrawalSlippage = 1e15;
+        maxDepositSlippage = 1e15;
+
+        IERC20[] memory poolAssets = getPoolAssets();
```

Comment:
Optional - I love the checking concept here, but seems weird that is in the aura abstract contract, because it's not related to aura specific functionality or storage slots. 

Same thing actually with the slippage. That's not aura specific either.

Reply:
good catch, initialize fits into BaseBalancerStrategy much better